### PR TITLE
feat: add HeavenlyDoubleHalberd weapon with multi-target kill

### PR DESCRIPTION
**Pull request by scolley31**
## Summary
- 方天畫戟 (EDQ103)：攻擊範圍 4，出殺時若該殺是最後一張手牌，可額外指定至多 2 名目標
- 三國殺 codebase 第一個多目標殺機制 — sequential polling（mirror ArrowBarrage pattern）
- 採用 dedicated API (`useHeavenlyDoubleHalberdKill`)，一次傳入 primary + additional targets
- 0 個 additional target → 短路走 `playerPlayCard`，等同正常殺（零 regression 保證）

### Domain
- `HeavenlyDoubleHalberdCard.java` — 武器卡 (range 4)
- `HeavenlyDoubleHalberdKillBehavior.java` — 繼承 `NormalActiveKillBehavior`，完全 override `playerAction()` + `doResponseToPlayerAction()`，用 `reactionPlayers` 列表 + `currentReactionPlayer` 做 sequential polling
- `HeavenlyDoubleHalberdKillTriggerEvent.java` — 通知前端方天畫戟發動 (含 attacker/cardId/targetPlayerIds)
- `Game.playerUseHeavenlyDoubleHalberdKill()` — 驗證 + 短路 + 推 behavior

### 既有檔案修改
- `DyingAskPeachBehavior.java`：
  - `JudgementIfRemoveBehavior` 排除 halberd（避免中間目標死亡時誤 pop 多目標 behavior）
  - 新增 `addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior` hook（瀕死解決後恢復下一目標的詢問）
- `EightDiagramTacticEquipmentEffectHandler.java`：新增 instanceof 分支，八卦陣成功後推進 halberd 到下一目標
- `BehaviorData.java`：新增 halberd deserialization case

### 瀕死中斷流程（重點！）
目標 C 在被殺時 HP=0 → 進入 DyingAskPeachBehavior → 所有人跳過桃 → C 永久死亡 → hook 恢復 halberd 並 emit AskDodgeEvent(D) → D 正常結算。**Domain test `testUseHalberdKill_MiddleTargetDying_DFlowStillResumes` 全程驗證此流程通過。**

## Test plan
- [x] `mvn test` in `domain/` — 353 tests pass (+18)
- [x] `mvn test` in `spring/` — 210 tests pass (+7)
- [x] `mvn install -DskipTests` from parent — 編譯全綠
- [x] 瀕死中斷 test 包含在 Phase 1

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/DyingAskPeachBehavior.java [hunk 1]
```diff
@@ -208,6 +214,18 @@ private void addAskDodgeEventIfCurrentBehaviorIsArrowBarrageBehavior(List<Domain
...
     }
 
+    private void addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior(List<DomainEvent> events) {
+        game.peekTopBehaviorSecondElement().ifPresent(secondBehavior -> {
+            if (secondBehavior instanceof HeavenlyDoubleHalberdKillBehavior halberdBehavior) {
+                Player halberdCurrentReactionPlayer = halberdBehavior.getCurrentReactionPlayer();
+                // halberd 在扣血進入瀕死時已將 currentReactionPlayer 推進到下一位
+                List<DomainEvent> dodgeOrEquipmentEvents = new ArrayList<>();
+                halberdBehavior.askCurrentTargetDodgeOrEquipmentEffect(dodgeOrEquipmentEvents);
+                events.addAll(dodgeOrEquipmentEvents);
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/DyingAskPeachBehavior.java:225]: **[Bug/Low]** 這個 hook 缺少一個 guard：如果 halberd 的 `isOneRound` 已為 true（最後一個目標已處理完），hook 不應該再發出 `AskDodgeEvent`。

**復現場景**：A 用方天畫戟殺 [B, C, D]，D 是最後一個目標。D 不出閃 → HP=0 → 進入 `DyingAskPeachBehavior`。所有人跳過桃 → D 永久死亡。此時：
- halberd 的 `isOneRound = true`（在 `doResponseToPlayerAction` 的 dying 分支 line 93-94 已設定）
- `currentReactionPlayer` 仍指向 D（因為 isLast=true 不呼叫 `advanceToNextTarget`）
- 這個 hook 呼叫 `askCurrentTargetDodgeOrEquipmentEffect` → 對已死亡的 D 發出 `AskDodgeEvent` → spurious event

**影響**：前端收到一個不應該出現的 `AskDodgeEvent(player-d)`，但因為 halberd behavior 已經 `isOneRound=true` 會被 pop 掉，所以沒有 handler 處理回應。功能上 harmless 但不乾淨。

**Fix**：加 guard
```java
private void addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior(List<DomainEvent> events) {
    game.peekTopBehaviorSecondElement().ifPresent(secondBehavior -> {
        if (secondBehavior instanceof HeavenlyDoubleHalberdKillBehavior halberdBehavior) {
            if (halberdBehavior.isOneRound()) return; // 已完成，不需要再問
            halberdBehavior.askCurrentTargetDodgeOrEquipmentEffect(new ArrayList<>());
            // ...
        }
    });
}
```

對比 ArrowBarrage 的 hook（line 206-214）用 `isInReactionPlayers` 做了等效的 guard。

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/HeavenlyDoubleHalberdKillBehavior.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * 八卦陣 handler 會呼叫此 method 取代 isOneRound 的判斷邏輯。
+     * 方天畫戟為多目標輪詢，八卦陣在中間目標成功並不代表整個 behavior 結束。
+     */
+    @Override
+    public boolean judgeWhetherRemoveTopBehavior() {
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/HeavenlyDoubleHalberdKillBehavior.java:131]: **[Observation / Good]** 👍 Override `judgeWhetherRemoveTopBehavior()` 回傳 false 是一個精準的設計。

`EightDiagramTacticEquipmentEffectHandler` 的 line 58-59 用 `judgeWhetherRemoveTopBehavior()` 決定是否在八卦陣成功後 pop topBehavior。對 `NormalActiveKillBehavior`（單目標殺）這應該回傳 true；但對 halberd（多目標輪詢）不應該在中間目標的八卦陣成功後就 pop。回傳 false 讓 EightDiagramTactic handler 的邏輯正確地保留 halberd 在 stack 上。

比 ArrowBarrage 的做法更乾淨 — ArrowBarrage 沒有 override 此 method（因為它本來就不是 `NormalActiveKillBehavior` 的子類），而是靠 `instanceof` 檢查走另一條路。

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/HeavenlyDoubleHalberdKillBehavior.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+     */
+    public boolean isLastReactionPlayer(String playerId) {
+        return reactionPlayers.get(reactionPlayers.size() - 1).equals(playerId);
+    }
+
+    /**
+     * 將 currentReactionPlayer 推進到 reactionPlayers 列表中的下一位。
+     * 注意：使用列表順序 (index + 1)，不是座位順序。
+     */
+    public void advanceToNextTarget() {
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/HeavenlyDoubleHalberdKillBehavior.java:146]: **[Nit]** `advanceToNextTarget()` 和 `askCurrentTargetDodgeOrEquipmentEffect()` 都是 `public`，但只被三個地方呼叫：
1. `doResponseToPlayerAction` (本 class)
2. `DyingAskPeachBehavior` hook
3. `EightDiagramTacticEquipmentEffectHandler` hook

由於 hooks 在不同 package 裡呼叫，public 是必要的。但如果未來重構成 polymorphic hook（e.g. `Behavior.onTargetDodgedByArmor(events)` / `Behavior.onDyingResolved(events)`），這些 method 可以收縮到 protected。

目前沒有可操作的修改建議，僅 flag 為未來 refactoring 機會。

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/DyingAskPeachBehavior.java [hunk 4]
```diff
@@ -173,7 +177,9 @@ private void JudgementIfRemoveBehavior() {
         Stack<Behavior> topBehavior = game.getTopBehavior();
         IntStream.range(0, topBehavior.size())
                 .mapToObj(i -> topBehavior.get(topBehavior.size() - 1 - i))
-                .filter(behavior -> behavior instanceof NormalActiveKillBehavior
+                .filter(behavior -> (behavior instanceof NormalActiveKillBehavior
+                        // 方天畫戟為多目標輪詢，中間目標死亡不應自動 pop
+                        && !(behavior instanceof HeavenlyDoubleHalberdKillBehavior))
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/DyingAskPeachBehavior.java:182]: **[Observation / Good]** 👍 在 `JudgementIfRemoveBehavior` 排除 `HeavenlyDoubleHalberdKillBehavior` 是防止 **中間目標死亡時 halberd 被提前 pop** 的關鍵。

如果不加這個排除，以下流程會壞掉：
1. A 用方天畫戟殺 [B, C, D]，B 已處理完
2. C 不出閃 → HP=0 → `DyingAskPeachBehavior` push
3. 所有人跳過桃 → C 永久死亡
4. `JudgementIfRemoveBehavior` 找到 stack 上的 halberd（是 `NormalActiveKillBehavior` 子類）
5. **沒有排除**：把 halberd 的 `isOneRound` 設 true → pop → D 永遠不會被詢問出閃
6. **有排除**：跳過 halberd → halberd 保留在 stack → D 正常被詢問

與 ArrowBarrage 的不同：ArrowBarrage 不是 `NormalActiveKillBehavior` 子類，所以 filter 本來就不會匹配到它。Halberd 繼承了 `NormalActiveKillBehavior`，所以需要顯式排除。

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/Game.java [hunk 5]
```diff
@@ -909,6 +910,92 @@ public List<DomainEvent> playerUseViperSpearKill(String playerId, String targetP
...
+        }
+
+        // 7. additional targets 驗證
+        List<String> additionalIds = additionalTargetPlayerIds == null ? new ArrayList<>() : additionalTargetPlayerIds;
+        if (additionalIds.size() > 2) {
+            throw new IllegalArgumentException("At most 2 additional targets");
+        }
+
+        // 8. 短路：沒有 additional target → 走正常 playCard 路徑
+        if (additionalIds.isEmpty()) {
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/Game.java:957]: **[Observation]** 短路到 `playerPlayCard` 的設計是好的 — 保證 0 additional 等同正常殺，零 regression。

注意：因為 step 5（handSize==1 check）在 step 8（短路）之前，即使 additional 為空也必須是最後一張手牌。這意味著擁有 2+ 手牌的玩家不能透過此 endpoint 打一般殺 — 必須用 `playerPlayCard` 端點。

這是正確的語意：`useHeavenlyDoubleHalberdKill` endpoint 專門給「方天畫戟效果觸發場景」使用。前端應在「殺是最後一張手牌 + 裝備方天畫戟」時才顯示此 UI。

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/effect/EightDiagramTacticEquipmentEffectHandler.java [hunk 6]
```diff
@@ -58,6 +59,7 @@ protected List<DomainEvent> doHandle(String playerId, String cardId, String targ
         topBehavior.setIsOneRound(isOneRoundBehavior && isEightDiagramTacticEffectSuccess);
         if (isEightDiagramTacticEffectSuccess) {
             addAskDodgeEventIfCurrentBehaviorIsArrowBarrageBehavior(domainEvents);
+            addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior(domainEvents, playerId);
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/effect/EightDiagramTacticEquipmentEffectHandler.java:62]: **[Observation]** `addAskDodgeEventIfCurrentBehaviorIsArrowBarrageBehavior` + `addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior` 兩個 `instanceof` 分支開始產生 code smell。

如果未來有第三種 multi-target behavior（例如自訂武將技能），這裡又要加一個 branch。更乾淨的做法是在 `Behavior` 加一個可 override 的 hook `onArmorDodgeSuccess(List<DomainEvent> events, String playerId)`:默認實現 → 不做任何事；ArrowBarrage / Halberd override → 推進到下一目標。

不阻塞本 PR。

File: LegendsOfTheThreeKingdoms/domain/src/test/java/com/gaas/threeKingdoms/HeavenlyDoubleHalberdTest.java [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+        Player playerB = game.getPlayer("player-b");
+        equipHalberdWithSingleKill(playerA);
+        playerB.getEquipment().setArmor(new EightDiagramTactic(ES2015));
+
+        List<DomainEvent> events = game.playerUseHeavenlyDoubleHalberdKill(
+                "player-a", BS8008.getCardId(), "player-b", List.of("player-c"));
+
+        assertTrue(events.stream().anyMatch(e -> e instanceof AskPlayEquipmentEffectEvent));
+        assertFalse(events.stream().anyMatch(e -> e instanceof AskDodgeEvent));
+    }
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/test/java/com/gaas/threeKingdoms/HeavenlyDoubleHalberdTest.java:193]: **[Observation / Good]** 👍 這是整個 PR 最有價值的 test — 完整驗證了瀕死中斷恢復流程：
1. A 用方天畫戟殺 [B, C, D]
2. B 出閃 → C 不出閃 → HP=0 → 瀕死
3. 5 人都跳過桃 → C 永久死亡
4. **驗證** D 收到 `AskDodgeEvent`（流程恢復）

這個 test 在 plan 階段被標記為 R1 風險，結果 TDD 直接通過了 — 證明 ArrowBarrage 的 hook pattern 可以正確延伸到 halberd。

File: LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/repository/data/BehaviorData.java [hunk 8]
```diff
@@ -150,6 +150,14 @@ yield new ViperSpearKillBehavior(
                         discardedCardIds
                 );
             }
+            case "HeavenlyDoubleHalberdKillBehavior" -> new HeavenlyDoubleHalberdKillBehavior(
+                    game,
+                    game.getPlayer(behaviorPlayerId),
+                    reactionPlayers,
+                    game.getPlayer(currentReactionPlayerId),
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/main/java/com/gaas/threeKingdoms/repository/data/BehaviorData.java:157]: **[Test Coverage]** 這段反序列化邏輯沒有對應的 unit test。建議在 `BehaviorDataTest.java` 補一個 round-trip test，驗證 `reactionPlayers` 多目標列表和 `currentReactionPlayerId` 在 serialize → deserialize 後完整保留。

（雖然 e2e test 的 multi-step 流程間接驗到了 MongoDB 持久化，但 unit test 更精準。）

**Review by scolley31 — COMMENTED** [review]
# Code Review: 方天畫戟 (HeavenlyDoubleHalberd) — 多目標殺

## 整體評價
codebase 第一個多目標殺機制，技術複雜度高。sequential polling 的設計正確 mirror ArrowBarrage，`JudgementIfRemoveBehavior` 排除 halberd 避免 mid-polling 誤 pop 是關鍵且正確的修正。瀕死中斷恢復流程通過 TDD 驗證（domain test `testUseHalberdKill_MiddleTargetDying_DFlowStillResumes`），證明設計能正確運作。

有一個低嚴重度的 edge case bug（最後一個目標死亡時 hook 發出 spurious AskDodge），和一些 test coverage gap 需要補。

## 缺少的測試場景
- [ ] 最後一個目標進入瀕死（last target dies）→ 確認 behavior 正確結束，不發出 spurious event
- [ ] 3 個目標的 e2e test（目前 e2e 只測了 2 個 additional target 的 trigger + all-skip）
- [ ] BehaviorData round-trip test（HeavenlyDoubleHalberdKillBehavior serialize → deserialize）

## 做得好的地方
- **瀕死中斷恢復**：`DyingAskPeachBehavior` 的 hook + `JudgementIfRemoveBehavior` 排除 halberd + domain test 三者一起確保 mid-polling dying 正確運作。這是整個 PR 最有價值的部分。
- **`judgeWhetherRemoveTopBehavior()` override 回傳 false**：巧妙地阻止了 `EightDiagramTacticEquipmentEffectHandler` 在中間目標八卦陣成功時把 halberd pop 掉。
- **短路設計**：0 additional targets → `playerPlayCard`，保證零 regression + 前端不需要額外處理。
- **ArrowBarrage 一致性**：dodge branch 用 `playerPlayCardNotUpdateActivePlayer`，正確把閃加到墓地，不像 `NormalActiveKillBehavior` 的 dodge branch 遺漏了 graveyard add。
- **Domain test 覆蓋 18 個 case**，包含裝備、觸發、全閃全跳、混合、八卦陣、瀕死恢復、10 個 validation。

## 總結

| 項目 | 評級 |
|------|------|
| 功能正確性 | 🟢 良好（mid-polling dying 通過 TDD，edge case bug 嚴重度低） |
| Pattern 一致性 | 🟢 良好（正確 mirror ArrowBarrage + EightDiagramTactic 慣例） |
| 程式碼品質 | 🟡 尚可（Game method 90 行 inline validation、instanceof 分支持續增長） |
| 測試品質 | 🟢 良好（18 domain + 7 e2e，瀕死 test 是亮點） |

**建議優先處理**：
1. **Hook guard for last-target-dies**：在 `addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior` 加 `isOneRound` 檢查
2. 補 domain test：最後一個目標進入瀕死並死亡
3. 補 BehaviorData round-trip test

**不阻塞 merge，可後續 PR 補**：
- DyingAskPeach / EightDiagramTactic 的 instanceof 分支增長（重構為 polymorphic hook）
- Game.playerUseHeavenlyDoubleHalberdKill 70+ 行 validation 抽 helper

🤖 Generated with [Claude Code](https://claude.com/claude-code)

**Review by scolley31 — COMMENTED** [review]
# Code Review v2: 方天畫戟 — review feedback fix

## 整體評價
第二個 commit `844a75a` 精準處理了 v1 review 的三個 feedback：`isOneRound` guard 修正 spurious event bug、last-target-dies domain test、BehaviorData round-trip test。所有 v1 review issues 已結案。

## v1 Review 問題追蹤

| v1 問題 | 狀態 |
|---------|------|
| Hook 缺 `isOneRound` guard → spurious AskDodge for dead last target | ✅ 已修 |
| 缺 domain test：最後目標進入瀕死 | ✅ 已補 |
| 缺 BehaviorData round-trip test | ✅ 已補 |
| EightDiagramTactic / DyingAskPeach instanceof 分支增長 | ⏳ 留到後續重構 |
| Game method 90 行 inline validation | ⏳ 留到後續重構 |

## 總結

| 項目 | v1 | v2 |
|------|----|----|  
| 功能正確性 | 🟢 良好 | 🟢 良好（spurious event 已修） |
| Pattern 一致性 | 🟢 良好 | 🟢 良好 |
| 程式碼品質 | 🟡 尚可 | 🟡 尚可（未變動） |
| 測試品質 | 🟢 良好 | 🟢 良好（+2 tests，gap 已補） |

**結論**：✅ **可以 merge**。所有功能性 feedback 已處理完畢。

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/DyingAskPeachBehavior.java [hunk 9]
```diff
@@ -208,6 +214,19 @@ private void addAskDodgeEventIfCurrentBehaviorIsArrowBarrageBehavior(List<Domain
         });
     }
 
+    private void addAskDodgeEventIfCurrentBehaviorIsHeavenlyDoubleHalberdKillBehavior(List<DomainEvent> events) {
+        game.peekTopBehaviorSecondElement().ifPresent(secondBehavior -> {
+            if (secondBehavior instanceof HeavenlyDoubleHalberdKillBehavior halberdBehavior) {
+                // 最後一個目標已處理完（isOneRound=true），不需要再問下一位
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/main/java/com/gaas/threeKingdoms/behavior/behavior/DyingAskPeachBehavior.java:220]: **[Observation / Good]** 👍 `if (halberdBehavior.isOneRound()) return;` — 精準修正了 v1 review 指出的 spurious AskDodge bug。

對比 ArrowBarrage 的 hook（line 206-214）用 `isInReactionPlayers` 做等效 guard：ArrowBarrage 的 `currentReactionPlayer` 在最後一人處理完後會被推進到 reactionPlayers 之外（via `getNextPlayer`），所以 `isInReactionPlayers` 回傳 false 起到 guard 作用。

Halberd 的 `advanceToNextTarget()` 不會推進到列表外（有 bounds check），所以用 `isOneRound()` 作為 guard 更直觀。兩種 guard 語意不同但效果一致。

File: LegendsOfTheThreeKingdoms/domain/src/test/java/com/gaas/threeKingdoms/HeavenlyDoubleHalberdTest.java [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+
+        // 所有人跳過桃 → D 永久死亡 (peach-ask: D → E → A → B → C)
+        game.playerPlayCard("player-d", "", "player-d", PlayType.SKIP.getPlayType());
+        game.playerPlayCard("player-e", "", "player-d", PlayType.SKIP.getPlayType());
+        game.playerPlayCard("player-a", "", "player-d", PlayType.SKIP.getPlayType());
+        game.playerPlayCard("player-b", "", "player-d", PlayType.SKIP.getPlayType());
+        List<DomainEvent> finalEvents =
+                game.playerPlayCard("player-c", "", "player-d", PlayType.SKIP.getPlayType());
+
+        // behavior stack 應為空（halberd 已結束）
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/domain/src/test/java/com/gaas/threeKingdoms/HeavenlyDoubleHalberdTest.java:257]: **[Observation / Good]** 這個 test 直接覆蓋了 v1 review 指出的 edge case：最後一個目標（D）不出閃 → HP=0 → 瀕死 → 所有人跳過桃 → D 死亡。

兩個 assertion 都很關鍵：
1. `assertEquals(0, game.getTopBehavior().size())` — halberd behavior 正確結束
2. `assertFalse(finalEvents.stream().anyMatch(e -> e instanceof AskDodgeEvent))` — 沒有 spurious AskDodge

如果沒有 `isOneRound()` guard，assertion 2 會失敗（D 收到不該有的 AskDodgeEvent）。

File: LegendsOfTheThreeKingdoms/spring/src/test/java/com/gaas/threeKingdoms/data/BehaviorDataTest.java [hunk 11]
```diff
@@ -287,4 +288,37 @@ public void testViperSpearKillBehaviorRoundTrip_PreservesDiscardedCardIds() {
...
+        assertEquals("HeavenlyDoubleHalberdKillBehavior", behaviorData.getBehaviorName());
+        assertEquals("BS8008", behaviorData.getCardId());
+        assertEquals(reactionPlayers, behaviorData.getReactionPlayers());
+        assertEquals("player-b", behaviorData.getCurrentReactionPlayerId());
+
+        assertInstanceOf(HeavenlyDoubleHalberdKillBehavior.class, restored);
+        HeavenlyDoubleHalberdKillBehavior restoredHalberd = (HeavenlyDoubleHalberdKillBehavior) restored;
+        assertEquals(reactionPlayers, restoredHalberd.getReactionPlayers());
+        assertEquals("player-b", restoredHalberd.getCurrentReactionPlayer().getId());
+        assertEquals("player-a", restoredHalberd.getBehaviorPlayer().getId());
```
Comment (scolley31) [inline LegendsOfTheThreeKingdoms/spring/src/test/java/com/gaas/threeKingdoms/data/BehaviorDataTest.java:321]: **[Observation / Good]** 這個 round-trip test 驗證了 halberd 的多目標狀態在 MongoDB 持久化後能正確還原。特別重要的 assertion：

- `assertEquals(reactionPlayers, restoredHalberd.getReactionPlayers())` — 多目標列表完整
- `assertEquals("player-b", restoredHalberd.getCurrentReactionPlayer().getId())` — 當前輪詢位置正確
- `assertInstanceOf(Kill.class, restoredHalberd.getCard())` — Kill 卡片型別正確（與 ViperSpear 的 VirtualKill 不同）

如果 `BehaviorData.createBehavior` 的 halberd case 有 bug（例如 `PlayCard.findById(cardId)` 找不到），這個 test 會 catch 到。
